packages/datetime/src/dateRangeInput.tsx

```diff
@@ -370,28 +410,45 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
     }
 
     private getSelectedRange = () => {
-        return [this.state.selectedStart, this.state.selectedEnd].map((selectedBound?: moment.Moment) => {
+        const momentDateRange = [this.state.selectedStart, this.state.selectedEnd] as MomentDateRange;
+        const [startDate, endDate] = momentDateRange.map((selectedBound?: moment.Moment) => {
             return this.isMomentValidAndInRange(selectedBound)
                 ? fromMomentToDate(selectedBound)
                 : undefined;
-        }) as DateRange;
+        });
+        // show only the start date if the dates overlap
+        // TODO: add different handling for the === case once
```

Comment:
TODO in a different PR, that is. Probably don't even need to keep this for now, but wanted to flag in the moment.